monai/data/dataset.py

```diff
@@ -45,10 +45,7 @@ def __init__(self, data, transform=None):
             transform (Callable, optional): transforms to execute operations on input data.
         """
         self.data = data
-        if isinstance(transform, Compose):
-            self.transform = transform
-        else:
-            self.transform = Compose(ensure_tuple(transform))
+        self.transform = transform
```

Comment:
this is nice simplification, need to update the docstring, and i'm surprised that there's no need to change any unit tests for this...

Reply:
What doc-string do you expect to update?
Actually, I just restored the code, Hans added the `Compose(ensure_tuple())` logic in the `PersistentDataset` related PR before. I feel it's unnecessary to do it in the base class.
Thanks.

Reply:
`transforms to execute operations` -> `a callable data transform`.
the class docstring could be `A generic dataset with a length property and an optional callable data transform when fetching a data sample for a given key.`

Reply:
updated in the latest patch.
Thanks.